src/main/java/com/rarchives/ripme/ripper/rippers/DeviantartRipper.java

```diff
@@ -1,409 +1,531 @@
+	private Map<String, String> cookies;
+	private DownloadThreadPool deviantartThreadPool = new DownloadThreadPool("deviantart");
+	private ArrayList<String> names = new ArrayList<String>();
+
+	// Constants
+	private final String referer = "https://www.deviantart.com/";
+	private final String userAgent = "Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:65.0) Gecko/20100101 Firefox/65.0";
+
+	@Override
+	public DownloadThreadPool getThreadPool() {
+		return deviantartThreadPool;
+	}
+
+	public DeviantartRipper(URL url) throws IOException {
+		super(url);
+	}
+
+	@Override
+	protected String getDomain() {
+		return "deviantart.com";
+	}
+
+	@Override
+	public String getHost() {
+		return "deviantart";
+	}
+
+	@Override
+	protected Document getFirstPage() throws IOException {
+		login();
+		return Http.url(urlWithParams(this.offset)).cookies(getDACookie()).referrer(referer).userAgent(userAgent).get();
+	}
+
+	/**
+	 * Stores logged in Cookies. Needed for art pieces only visible to logged in
+	 * users.
+	 * 
+	 * 
+	 * @throws IOException when failed to load webpage or failed to read/write
+	 *                     cookies in file (used when running multiple instances of
+	 *                     RipMe)
+	 */
+	private void login() throws IOException {
+
+		File f = new File("DACookie.toDelete");
+		if (!f.exists()) {
+			f.createNewFile();
+			f.deleteOnExit();
+
+			// Load login page
+			Response res = Http.url("https://www.deviantart.com/users/login").connection().method(Method.GET)
+					.referrer(referer).userAgent(userAgent).execute();
+
+			// Find tokens
+			Document doc = res.parse();
+			Element form = doc.getElementById("login");
+			String token = form.select("input[name=\"validate_token\"]").first().attr("value");
+			String key = form.select("input[name=\"validate_key\"]").first().attr("value");
+			System.out.println(
+					"------------------------------" + token + "   &   " + key + "------------------------------");
+
+			// Build Login Data
+			HashMap<String, String> loginData = new HashMap<String, String>();
+			loginData.put("challenge", "");
+			loginData.put("username", username);
+			loginData.put("password", password);
+			loginData.put("remember_me", "1");
+			loginData.put("validate_token", token);
+			loginData.put("validate_key", key);
+			Map<String, String> cookies = res.cookies();
+
+			// Log in using data. Handle redirect
+			res = Http.url("https://www.deviantart.com/users/login").connection().referrer(referer).userAgent(userAgent)
+					.method(Method.POST).data(loginData).cookies(cookies).followRedirects(false).execute();
+			this.cookies = res.cookies();
+
+			res = Http.url(res.header("location")).connection().referrer(referer).userAgent(userAgent)
+					.method(Method.GET).cookies(cookies).followRedirects(false).execute();
+
+			// Store cookies
+			updateCookie(res.cookies());
+
+			// Apply agegate
+			this.cookies.put("agegate_state", "1");
+
+			// Write Cookie to file for other RipMe Instances
```

Comment:
Why write a file to disk here? You could just set a value in ripmes config file. There's Utils.setConfigString("DeviantartRipper.cookies", cookies.toString()) for setting it and AFAIK there is another util func for turning cookie strings back into cookies

Reply:
Login Cookie is now stores as BASE64 encoded String in config file.